src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -499,18 +499,27 @@ trait GoldTransforms extends SparkSessionWrapper {
                                    clsfIncrementalDF : DataFrame,
                                  )(auditIncrementalDF: DataFrame): DataFrame = {
 
+    if (auditIncrementalDF.isEmpty) {
+      throw new NoNewDataException("No New Data", Level.WARN, true)
+    }
     val auditDF_base = auditIncrementalDF
       .filter(col("serviceName") === "notebook" && col("actionName") === "runCommand")
       .selectExpr("*", "requestParams.*").drop("requestParams")
 
     if (auditDF_base.columns.contains("executionTime")){
+      if (notebook.asDF.isEmpty){
+        throw new NoNewDataException("No New Data",Level.WARN, true)
+      }
       val notebookLookupTSDF = notebook.asDF
         .select("organization_id", "notebook_id", "notebook_path", "notebook_name", "unixTimeMS", "date")
         .withColumnRenamed("notebook_id", "notebookId")
         .filter('notebookId.isNotNull)
         .distinct
         .toTSDF("unixTimeMS", "organization_id", "date", "notebookId")
 
+      if (clsfIncrementalDF.isEmpty) {
+        throw new NoNewDataException("No New Data", Level.WARN, true)
+      }
```

Comment:
Please add all the empty check in the beginning of the function.I suggest create a function to check the No new data exception and call that function in the beginning.

Reply:
Done